src/shadowbox/server/manager_service.ts

```diff
@@ -224,22 +229,22 @@ export class ShadowsocksManagerService {
   }
 
   public getShareMetrics(req: RequestType, res: ResponseType, next: restify.Next): void {
-    res.send(200, {metricsEnabled: this.metricsPublisher.isSharingEnabled()});
+    res.send(Success.OK, {metricsEnabled: this.metricsPublisher.isSharingEnabled()});
     next();
   }
 
   public setShareMetrics(req: RequestType, res: ResponseType, next: restify.Next): void {
-    const params = req.params as SetShareMetricsParams;
-    if (typeof params.metricsEnabled === 'boolean') {
-      if (params.metricsEnabled) {
-        this.metricsPublisher.startSharing();
-      } else {
-        this.metricsPublisher.stopSharing();
-      }
-      res.send(204);
+    if (!req.params || typeof req.params.metricsEnabled !== 'boolean') {
```

Comment:
I would leave in the `params` variable with the cast. It makes it explicit that you're dealing with a `SetShareMetricsParam` object and you don't have to keep accessing the request.

Reply:
Sure.  the reason i wanted to get rid of the cast originally was because it seemed to imply that there's some type safety gained there, but it doesn't do any runtime type checks or anything like that.